packages/core/src/event_emitter.ts

```diff
@@ -115,57 +115,31 @@ class EventEmitter_ extends Subject<any> {
   }
 
   subscribe(observerOrNext?: any, error?: any, complete?: any): Subscription {
-    let schedulerFn: (t: any) => any;
-    let errorFn = (err: any): any => null;
-    let completeFn = (): any => null;
-
-    if (observerOrNext && typeof observerOrNext === 'object') {
-      schedulerFn = this.__isAsync ? (value: any) => {
-        setTimeout(() => observerOrNext.next(value));
-      } : (value: any) => {
-        observerOrNext.next(value);
-      };
-
-      if (observerOrNext.error) {
-        errorFn = this.__isAsync ? (err) => {
-          setTimeout(() => observerOrNext.error(err));
-        } : (err) => {
-          observerOrNext.error(err);
-        };
+    const isObserver = observerOrNext && typeof observerOrNext === 'object';
+    let schedulerFn = isObserver ?
+        (observerOrNext as PartialObserver<any>).next?.bind(observerOrNext) :
+        observerOrNext;
+    let errorFn =
+        (isObserver ? (observerOrNext as PartialObserver<any>).error?.bind(observerOrNext) :
+                      error) ||
+        (() => null);
+    let completeFn = isObserver ?
+        (observerOrNext as PartialObserver<any>).complete?.bind(observerOrNext) :
+        complete;
```

Comment:
I'd propose to have an if block for `isObserver` branch (mostly for readability):
```suggestion
    let nextFn = observerOrNext;
    let errorFn = error || (() => null);
    let completeFn = complete;
    if (observerOrNext && typeof observerOrNext === 'object') {
      const observer = observerOrNext as PartialObserver<unknown>;
      nextFn = observer.next?.bind(observer);
      errorFn = observer.error?.bind(observer);
      completeFn = observer.complete?.bind(observer);
    }
```